### ISSUE TITLE
Monitoring - Alertmanager is not working when TLS enabled

### ISSUE THREAD

**Issue by MKlimuszka** [rancher/rancher#40307]
Rancher Cluster:
Rancher version: 2.6.5
Number of nodes: 3
Node OS version: Ubuntu

Downstream Cluster:
Number of Downstream clusters: 2
Node OS: Ubuntu
RKE/RKE2/K3S version: RKE
Kubernetes version: 
CNI:

Longhorn:
Longhorn version:
CPU per node:
Memory per node:
Disk type: HDD/SSD/NVMe
Network bandwidth between the nodes:

Other:
Underlying Infrastructure: AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal VMware
SURE-4788

ssue description:
Customer wants to use the Alert manager with TLS enabled ( Including username and password). 

Business impact:
Unable to use monitoring

Troubleshooting steps:
We have enabled monitoring using the Rancher chart and configured the Email.

    We have enabled Legacy monitoring for checking : 

    For smtp relay (no authentication needed) alerts works
    For smtp (with authentication)
    “Test” option works (we received the email)
    BUT Alarms are not arriving

    Alert monitor (new monitoring):
        “Test” option doesn’t exist, so we cannot check it
        Alarms are not arriving
    For smtp relay (no authentication needed) alerts works
    For smtp (with authentication)

To make sure no blocking to the ports and authentication works fine,  we sent the email manually as below

        Email using authentication from a KaaS server and it is working

Repro steps:

Rancher 2.6.5

Install the latest Monitoring chart from Apps/Market place

Configure Receiver as Email and Routes

Use the TLS enabled and provide the SMTP relay user name and password.

For testing enabling Legacy and configure the Alerts and do a test and its sending the emails but when configuring the Routes its not sending.

Workaround:
Is work around available and implemented? NO
What is the workaround: 

Actual behavior:
Email Alerting is not working when the Authentication is enabled. 

Expected behaviour:
Email Alerting should work when the Authentication is enabled. 

See original report for additional logs and info

**Comment by MKlimuszka** [comment]
This needs to be reproduced in 2023-Q1-2.6x and 2023-Q1-2.7x, and if it can't be reproduced in those, then 2.6.5

Prachi's last comment: "AlertManager with TLS enabled is supported, but possibly this needs documentation for clarity around how to configure it. We need to test this out in house and document how to configure Alerts with TLS enabled using monitoring v2. "

**Comment by nflynt** [comment]
Running through all the relevant TLS configurations against both a local postfix server and a remote Fastmail SMTP endpoint, I've been unable to reproduce issues with TLS authentication in `2.7-head` or `2.6-head`. The UI issue that I discovered was a regression introduced after this bug report, so it was ultimately unrelated.

I did discover that "Use TLS" enforces the "STARTTLS" extension, and this leads to a counter-intuitive requirement to uncheck "Use TLS" for remote SMTP endpoints that perform implicit TLS on port 465. This is [working as designed](https://prometheus.io/docs/alerting/latest/configuration/#configuration-file) from AlertManager's point of view, so it's a labeling and documentation issue on the Rancher side which I will file separately.

I *was* able to reproduce several TLS-related issues on `v2.6.5`, mostly UI bugs that affect saving of configuration and secrets, and a chart-level bug that caused the "Use TLS" setting to be ignored. These issues have all been patched, and as `v2.6.5` is no longer supported, I'm recommend affected users to upgrade.
